### DISCUSSION TITLE
Labels <-> Tags ?

### DISCUSSION THREAD

**Discussion by gotjoshua** [zxch3n/PomodoroLogger#243] [Q&A]
I am starting to implement sync with gitlab, and i want to add labels to kanban cards.
I found that there is already the concept of a Tag and a TagManager, but i don't see how to use it in the ui.

Is it a partially implemented feature?

edit: just saw this issue https://github.com/zxch3n/PomodoroLogger/issues/242 and realized that tags are parsed from the description.

I'd like to think together how to improve the data model to be compatible with gitlab labels, or to consciously choose to let the parsed description  #tags coexist with a more robust Label system...

**Comment by zxch3n** [comment]
Good idea~ We can add a new field `label` to the card. And create a label manager for each board. And inside each description, we can refer to label / other cards by `#label` syntax. And tags are the same as labels, in my opinion. I rarely see a product introduce both concepts with different meanings.

**Reply by gotjoshua** [reply]
great, i will do that as part of my gitlab integration and mirror the gitlab data structure for labels onto the PomLog card type.

also will include [the amazing "scoped::labels" feature](https://docs.gitlab.com/ee/user/project/labels.html#scoped-labels)

**Reply by gotjoshua** [reply]
> label manager for each board

I'm not sure how to deal with this (both on the PomLog only level, nor on the gitlab integration level)

gitlab has nested inheritance on labels, which is very handy.
Actually i would prefer to be able to set up labels in various contexts and merge them selectively.

Example:
I have user labels that show up on all boards
I have organization labels that show up on all boards in that org
I have (sub)project labels that only show up downstream

**Reply by gotjoshua** [reply]
In progress:

https://github.com/onezoomin/PomodoroLogger/issues/5
